src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Metadata/DefaultModelMetadataProvider.cs

```diff
@@ -232,15 +252,22 @@ protected virtual DefaultMetadataDetails[] CreatePropertyDetails(ModelMetadataId
         /// </remarks>
         protected virtual DefaultMetadataDetails CreateTypeDetails(ModelMetadataIdentity key)
         {
-            return new DefaultMetadataDetails(key, ModelAttributes.GetAttributesForType(key.ModelType));
+            ModelAttributes attributes;
+
+            if (key.MetadataKind == ModelMetadataKind.Parameter)
+            {
+                attributes = ModelAttributes.GetAttributesForParameter(key);
+            }
+            else
+            {
+                attributes = ModelAttributes.GetAttributesForType(key.ModelType);
+            }
+
+            return new DefaultMetadataDetails(key, attributes);
         }
 
         private class TypeCache : ConcurrentDictionary<ModelMetadataIdentity, ModelMetadataCacheEntry>
         {
-            public TypeCache()
```

Comment:
I removed all the stuff from this class only because it looks like it would behave the same without it, since `ModelMetadataIdentity` implements `GetHashCode` and `Equals` with pretty much the same logic. The tests all do still pass. If I'm missing something about why these extra custom implementation makes a difference, please let me know!

Reply:
I think it's probably fine